### ISSUE TITLE
Taking a mean over softmax dimension for LHP threshold

### ISSUE THREAD

**Issue by gurdeepmaurya** [hynnsk/HP#10]
Hi Hyun Seok,

If I'm correct, this line finds the threshold for LHP and sets all affinities below the "mean" affinity score to 0. 

https://github.com/hynnsk/HP/blob/3e92daa996ce3a4f418a4595d7f15c2d8ec9de80/model/dino/DinoFeaturizer.py#L171C31-L171C31

Since the shape of  `attn` at line 169 is [B, N, N] and line 169 performs softmax over dim=-1, i.e. dim=2, this implies

`torch.mean(attn, dim=2, keepdim=True)`  equals `self.sigma/N` 

because softmax dim sums to 1.0,  (self.sigma is a consequence of Line 170.)

I'd be very glad if you could clarify this.

Regards
Akshit Maurya

**Comment by hynnsk** [comment]
Hello,

I appreciate your interest, and I apologize for any confusion caused by the code implementation.

To clarify, the expression `torch.mean(attn, dim=2, keepdim=True)` is indeed equivalent to `self.sigma/N`.

The multiplication by sigma is done for enhanced visualization. Whether this multiplication is included or not, any values in the attention scores (H*W with a sum of 1) that are below the average value will be set to zero.

Thank you.

**Comment by gurdeepmaurya** [comment]
Thanks, @hynnsk, for a (very) prompt reply. 
Take Care!!
